src/box/iproto.cc

```diff
@@ -213,6 +225,13 @@ struct iproto_thread {
 		size_t requests_in_progress;
 		/** Iproto thread stat collected in tx thread. */
 		struct rmean *rmean;
+		/**
+		 * Connections drop is finished. That is all connections marked
+		 * to be dropped are dropped.
+		 */
+		bool drop_finished;
+		/** Condition for drop_finished. */
+		fiber_cond drop_cond;
```

Comment:
Nit: `struct fiber_cond`. Also, better call it `drop_finished_cond`.

Reply:
Done. Also moved flag and condition to the global level.

<details>
<summary>diff</summary>

```diff
diff --git a/src/box/iproto.cc b/src/box/iproto.cc
index 7bf81d242..28320b355 100644
--- a/src/box/iproto.cc
+++ b/src/box/iproto.cc
@@ -225,16 +225,14 @@ struct iproto_thread {
 		size_t requests_in_progress;
 		/** Iproto thread stat collected in tx thread. */
 		struct rmean *rmean;
-		/**
-		 * Connections drop is finished. That is all connections marked
-		 * to be dropped are dropped.
-		 */
-		bool drop_finished;
-		/** Condition for drop_finished. */
-		fiber_cond drop_cond;
 	} tx;
 };
 
+/** Condition for drop finished. */
+static struct fiber_cond drop_finished_cond;
+/** Count of iproto threads that are not finished connections drop yet. */
+static size_t drop_pending_count;
+
 /**
  * IPROTO listen URIs. Set by box.cfg.listen.
  */
@@ -1526,10 +1524,10 @@ iproto_connection_new(struct iproto_thread *iproto_thread)
 static void
 tx_process_drop_finished(struct cmsg *m)
 {
-	struct iproto_thread *iproto_thread =
-		container_of(m, struct iproto_thread, drop_finished_msg);
-	iproto_thread->tx.drop_finished = true;
-	fiber_cond_signal(&iproto_thread->tx.drop_cond);
+	(void)m;
+	assert(drop_pending_count > 0);
+	if (--drop_pending_count == 0)
+		fiber_cond_signal(&drop_finished_cond);
 }
 
 /** Send message to TX thread to notify that connections drop is finished. */
@@ -3293,7 +3291,6 @@ iproto_thread_init(struct iproto_thread *iproto_thread)
 	iproto_thread->tx.requests_in_progress = 0;
 	iproto_thread->requests_in_stream_queue = 0;
 	rlist_create(&iproto_thread->connections);
-	fiber_cond_create(&iproto_thread->tx.drop_cond);
 }
 
 /** Initialize the iproto subsystem and start network io thread */
@@ -3315,6 +3312,7 @@ iproto_init(int threads_count)
 	evio_service_create(loop(), &tx_binary, "tx_binary", NULL, NULL);
 	iproto_threads = (struct iproto_thread *)
 		xcalloc(threads_count, sizeof(struct iproto_thread));
+	fiber_cond_create(&drop_finished_cond);
 
 	for (int i = 0; i < threads_count; i++, iproto_threads_count++) {
 		struct iproto_thread *iproto_thread = &iproto_threads[i];
@@ -3578,6 +3576,7 @@ iproto_drop_connections(void)
 {
 	static struct latch latch = LATCH_INITIALIZER(latch);
 	latch_lock(&latch);
+	drop_pending_count = iproto_threads_count;
 	for (int i = 0; i < iproto_threads_count; i++) {
 		struct iproto_cfg_msg *cfg_msg =
 			(struct iproto_cfg_msg *)xmalloc(sizeof(*cfg_msg));
@@ -3586,9 +3585,8 @@ iproto_drop_connections(void)
 	}
 
 	for (int i = 0; i < iproto_threads_count; i++) {
-		while (!iproto_threads[i].tx.drop_finished)
-			fiber_cond_wait(&iproto_threads[i].tx.drop_cond);
-		iproto_threads[i].tx.drop_finished = false;
+		while (drop_pending_count != 0)
+			fiber_cond_wait(&drop_finished_cond);
 	}
 	latch_unlock(&latch);
 }
@@ -3816,6 +3814,7 @@ iproto_free(void)
 		iproto_req_handler_delete(handler);
 	}
 	mh_i32ptr_delete(tx_req_handlers);
+	fiber_cond_destroy(&drop_finished_cond);
 
 	/*
 	 * Here we close sockets and unlink all unix socket paths.

```

</details>